maat-scheduled-tasks/src/main/java/uk/gov/justice/laa/maat/scheduled/tasks/mapper/BillingDataFeedLogMapper.java

```
@@ -0,0 +1,58 @@
+import uk.gov.justice.laa.maat.scheduled.tasks.dto.BillingDTO;
+import uk.gov.justice.laa.maat.scheduled.tasks.dto.RepOrderBillingDTO;
+import uk.gov.justice.laa.maat.scheduled.tasks.entity.BillingDataFeedLogEntity;
+import uk.gov.justice.laa.maat.scheduled.tasks.enums.BillingDataFeedRecordType;
+
+@Component
+@RequiredArgsConstructor
+public class BillingDataFeedLogMapper {
+    private final ObjectMapper objectMapper;
+
+    public <T extends BillingDTO> BillingDataFeedLogEntity mapDtoToEntity(BillingDataFeedRecordType recordType, List<T> dtos)
+        throws JsonProcessingException {
+        if (dtos == null || dtos.isEmpty()) {
+            throw new IllegalArgumentException("Data to be serialised must be provided");
+        }
+
+        return BillingDataFeedLogEntity.builder()
+            .recordType(recordType.getValue())
+            .dateCreated(LocalDateTime.now())
+            .payload(objectMapper.writeValueAsString(dtos))
+            .build();
+    }
+
+    public List<ApplicantBillingDTO> mapEntityToApplicantBillingDtos(BillingDataFeedLogEntity entity) {
+        try {
+            return objectMapper.readValue(entity.getPayload(), new TypeReference<>() {});
+        } catch (JsonProcessingException | IllegalArgumentException e) {
+            return null;
+        }
+    }
+
+    public List<ApplicantHistoryBillingDTO> mapEntityToApplicationHistoryBillingDtos(BillingDataFeedLogEntity entity) {
+        try {
+            return objectMapper.readValue(entity.getPayload(), new TypeReference<>() {});
+        } catch (JsonProcessingException e) {
+            return null;
+        }
+    }
+
+    public List<RepOrderBillingDTO> mapEntityToRepOrderBillingDtos(BillingDataFeedLogEntity entity) {
+        try {
+            return objectMapper.readValue(entity.getPayload(), new TypeReference<>() {});
+        } catch (JsonProcessingException e) {
+            return null;
+        }
+    }
```

Can we log if any `JsonProcessingException`s are thrown here? Right now I think we just fail silently

---

Good point, thanks. In theory there shouldn't be any issues as we are writing out the JSON from the service but it's worth checking anyway 👍